src/Parser/Priest/Shadow/Modules/Spells/Voidform.js

```diff
@@ -40,136 +36,158 @@ class Voidform extends Analyzer {
     return Object.keys(this._voidforms).map(key => this._voidforms[key]);
   }
 
+  get nonExcludedVoidforms(){
+    return this.voidforms.filter(voidform => !voidform.excluded);
+  }
+
+  get averageVoidformStacks() {
+    if (this.voidforms.length === 0) return 0;
+    // ignores last voidform if seen as skewing
+    return this.nonExcludedVoidforms.reduce((p, c) => p += c.stacks.length, 0) / this.nonExcludedVoidforms.length;
+  }
+
   get averageVoidformHaste() {
-    const averageHasteFromVoidform = (this.voidforms.reduce((p, c) => p += c.totalHasteAcquired / ((c.ended - c.start) / 1000), 0) / this.voidforms.length) / 100;
-    return (1 + this.combatants.selected.hastePercentage) * (1 + averageHasteFromVoidform);
+    if(!this.currentVoidform) return (1 + this.combatants.selected.hastePercentage);
+    const averageHasteGainedFromVoidform = (this.voidforms.reduce((total, voidform) => total += voidform.averageGainedHaste, 0)) / this.voidforms.length;
```

Comment:
Here again I think this line could be shortened by moving totalVoidFormHaste or totalHaste